### DISCUSSION TITLE
Persist "Hide whitespace" preference across PRs

### DISCUSSION THREAD

**Discussion by madisonscott** [community/community#65597] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

Hey team, thanks for all your hard work! 

What are your thoughts on persisting the "Hide whitespace" preference in diff view settings across PRs, as the Unified/Split preference is persisted? I pretty much turn "Hide whitespace" on for every diff I look at, so it'd be nice to only ever do that once :)

Upon searching for similar discussions, I found an older thread that mentioned that this setting is intended to store your preference per-diff. To maintain that functionality, "Hide whitespace" could be added as a user-level setting, defaulting to that value when looking at a PR for the fist time, then still allowing it to be set a t a per-diff level on the diff itself.

Thanks for considering!

![image](https://github.com/community/community/assets/10393950/441af11a-4fd5-4905-827e-908838442e96)

**Comment by alexander-dermicus** [comment]
Note that this proposal already has a lot of support in the older thread: https://github.com/orgs/community/discussions/5486

**Comment by EskeRahn** [comment]
Personally I believe it would be even better if it was possible to set per filetype, as not ALL filetypes has whitespaces as pure cosmetics.

But ignoring whitespaces should also be coupled with not seeing solo { and } with a particular indentation as a good match for synchronisation. that causes bad syncs for e.g. C# and similar.

**Comment by ardicoetzee** [comment]
Yes, please add this, at least on a per-user level.

**Comment by mlafleur** [comment]
At a minimum, the "Show Whitespace" toolbar button should always be visible. It shouldn't be markedly easier to show whitespace than to hide it. 

![image](https://github.com/community/community/assets/3802472/b8af7bed-fcf4-4f16-a8a1-f512d619070d)

**Reply by pcnova** [reply]
THIS HERE! 🙏🏼

**Comment by cupcakearmy** [comment]
Reviewing code is a BIG part of the time i spend on github. Hiding whitespace is such a vital feature, i don't unterstand how this cannot be a priority. You already have the functionality, just persist it 🙏

**Comment by NicholasMTElliott** [comment]
My workflow literally involves manually turning on this flag on every single PR I do, because the vast majority of files I review the white-space does not matter.

**Comment by patrickholley-asurion** [comment]
It's amazing that a website literally for computer programmers can cause so much disruption in a programmer's flow. Please add this as a persistent option, it is annoying and slows me down having to check it every time.

**Comment by Rylab** [comment]
One more vote to please oh please make this a persistent user-level (and/or repo-level) setting! It's quite rare that whitespace changes matter in our workflows; swapping the default to ignore it, would save so much time.

**Comment by ardicoetzee** [comment]
A friend directed me to this as an interim solution: https://chromewebstore.google.com/detail/git-well-soon/ehpeaofieafibmhiagianfjjblpnmbdo

I.e. A chrome plugin that does exactly what we want!

**Reply by alexander-dermicus** [reply]
Great workaround! Thanks for sharing!

**Reply by HarryCruse-Kaluza** [reply]
It wants permission to "Read and change all your data on all websites". Not sure why it would need such a huge amount of access

**Comment by atkeranov** [comment]
A dev could literally implement this in a day at max. Now imagine how many human work hours are wasted by developers coming here to read through this thread, vote, and comment.

**Reply by EskeRahn** [reply]
My guess would be that it is not the work load of the hack that is the issue....

IMHO implementing it as suggested surely would be better than what we have currently for many (including me), but really would be a inadequate hack/workaround for a deeper problem in  
a) distinguishing logic and cosmetic changes, and   
b) finding good ways to display these differences.  

For the code part of say a .cs file spaces/indentions would be purely cosmetic, **but** inside a string value in the same file they might be important, with undesired nasty side effects, and thus _not_ ignorable. and really ought to be visible to a reviewer, to catch a crucial system breaking typo!

So to do a _logical_ diff without risk of false positives or false negatives some 'compiling' would be needed to decode each file into its _logical_ components, including first detecting its programming language correctly (relying on the file extension might not be enough), before a compare can be _safely_ done.
Maybe by the help af a compiler / de-compiler pair??

...And next comes thinking up a clever way of actually displaying the found _logical_ differences between two files that char by char seems very different, but are close to logically identical (as we currently have it with spaces, but could be different line splittings as well)

So certainly not something pulled off in a single day....

**Reply by drwharris** [reply]
> My guess would be that it is not the work load of the hack that is the issue....
> 
> IMHO implementing it as suggested surely would be better than what we have currently for many (including me), but really would be a inadequate hack/workaround for a deeper problem in a) distinguishing logic and cosmetic changes, and b) finding good ways to display these differences.
> 
> For the code part of say a .cs file spaces/indentions would be purely cosmetic, **but** inside a string value in the same file they might be important, with undesired nasty side effects, and thus _not_ ignorable. and really ought to be visible to a reviewer, to catch a crucial system breaking typo!
> 
> So to do a _logical_ diff without risk of false positives or false negatives some 'compiling' would be needed to decode each file into its _logical_ components, including first detecting its programming language correctly (relying on the file extension might not be enough), before a compare can be _safely_ done. Maybe by the help af a compiler / de-compiler pair??
> 
> ...And next comes thinking up a clever way of actually displaying the found _logical_ differences between two files that char by char seems very different, but are close to logically identical (as we currently have it with spaces, but could be different line splittings as well)
> 
> So certainly not something pulled off in a single day....

The functionality exists already. The only thing being asked is for a users to be able to have their preference to have it ON by default be remembered.

**Reply by EskeRahn** [reply]
@drwharris, Did you even read what I wrote, and what it was a comment to?

**Reply by drwharris** [reply]
I did and you have a valid point, however I use this existing functionality a lot and I use it knowing these limitations. And while a small inconvenience, I would like the option to default it to on.

**Reply by EskeRahn** [reply]
So you might have noticed that I started by saying that it for me too would be better than what we got now....

**Comment by johngallagher** [comment]
Please implement this. The previous thread has many comments. I heard devs complaining about this 3 years ago. It's 2024 and you have the backing of Microsoft. Please implement this.

**Comment by nelsondude** [comment]
This would be amazing to have! How can we get this feature prioritized? I find myself reviewing large PRs way too often, only to realize that its 95% whitespace changes. My company has strict policies on chrome extensions so that is not a viable option unfortunately.

**Comment by rgallagherab** [comment]
+1 on request for this user-level setting to be added.

**Comment by erorcodes** [comment]
This is one of the most annoying _daily_ things I do (and that says a LOT) so please please prioritize 🙏

**Comment by armadsen** [comment]
Yet another request for this. I've written my own AppleScript to add `?w=1` to the URL with a keyboard shortcut in Safari, but it sure would be nice if persisting this were a GitHub feature.

(AppleScript below if anyone wants it. I use the excellent [FastScripts](https://redsweater.com/fastscripts/) to give it a keyboard shortcut.)

```applescript
tell application "Safari"
	set currentDoc to front document
	set theWindow to window 1 where its document is currentDoc
	tell (current tab of theWindow)
		set URL to URL & "?w=1"
	end tell
end tell
```

**Comment by taurus227** [comment]
Another workaround is to use the [Requestly](https://requestly.com/) browser extension to automatically add `?w=1` to the end of the relevant github.com URLs. 

Here's my rule file that can be Imported into Requestly:

```json
[
    {
        "creationDate": 1586890092492,
        "description": "Hide whitespace changes",
        "groupId": "",
        "id": "QueryParam_1586890092492",
        "isSample": false,
        "name": "Whitespace",
        "objectType": "rule",
        "pairs": [
            {
                "modifications": [
                    {
                        "actionWhenParamExists": "Overwrite",
                        "param": "w",
                        "type": "Add",
                        "value": "1"
                    }
                ],
                "source": {
                    "filters": {},
                    "key": "Url",
                    "operator": "Wildcard_Matches",
                    "value": "*github.com/*/files*"
                }
            },
            {
                "modifications": [
                    {
                        "actionWhenParamExists": "Overwrite",
                        "param": "w",
                        "type": "Add",
                        "value": "1"
                    }
                ],
                "source": {
                    "filters": {},
                    "key": "Url",
                    "operator": "Wildcard_Matches",
                    "value": "*github.com/*/files/*..*"
                }
            },{
                "modifications": [
                    {
                        "actionWhenParamExists": "Overwrite",
                        "param": "w",
                        "type": "Add",
                        "value": "1"
                    }
                ],
                "source": {
                    "filters": {},
                    "key": "Url",
                    "operator": "Wildcard_Matches",
                    "value": "*github.com/*/commits/*"
                }
            },
            {
                "modifications": [
                    {
                        "actionWhenParamExists": "Overwrite",
                        "param": "w",
                        "type": "Add",
                        "value": "1"
                    }
                ],
                "source": {
                    "filters": {},
                    "key": "Url",
                    "operator": "Wildcard_Matches",
                    "value": "*github.com/*/commit/*"
                }
            },
            {
                "modifications": [
                    {
                        "actionWhenParamExists": "Overwrite",
                        "param": "w",
                        "type": "Add",
                        "value": "1"
                    }
                ],
                "source": {
                    "filters": {},
                    "key": "Url",
                    "operator": "Wildcard_Matches",
                    "value": "*github.com/*/compare/*"
                }
            }
        ],
        "ruleType": "QueryParam",
        "status": "Active"
    }
]
```

**Reply by madisonscott** [reply]
Thanks for sharing! This feels safer to me than the other extension going around :)

**Reply by sansari** [reply]
+1 this worked great for me, thanks for sharing!

**Comment by rbirkgit** [comment]
I cannot believe we still don't have it. Every day, every review, I have to manually enable hide whitespace. It's so annoying!

**Reply by HBoxer** [reply]
+1 please implement this.

**Comment by evan-bluecharm** [comment]
Would really like this feature, please!

GitKraken has it, so why not GitHub? :)

**Reply by luchillo17** [reply]
Because they're different companies with different codebases, just like GH has PR change suggestions that you can apply but BitBucket has not 😏

But honestly GH already has PR level whitespace hiding toggle, doubt it's that hard to add a global one that works as a default for the PR level...

**Comment by sansari** [comment]
Been waiting for this feature for years. Please add it!

**Comment by kevindflynn** [comment]
I continue to get halfway through a PR review, to scroll back to the top, to check "Hide whitespace changes" - and then scroll back down to where I was in my review.

**Comment by luchillo17** [comment]
It's been years since the original request was made, and 1 year since this discussion was made, lots of attention (almost 225 at this point), and yet no user level solution, is there anyone we can tag that could give us a status update on this?

**Comment by tuliolima** [comment]
It would be very useful. I am migrating from Gitlab and missing this feature.

**Comment by Aynekko** [comment]
How is this still not implement after 5 years of people asking? It's the end of 2024, please add it... It should be really a simple fix guys!

**Comment by DamonDisney** [comment]
Adding to the choir

**Comment by joaopedrodcf** [comment]
Please add this feature is such a life improvement

**Comment by PeteHanner** [comment]
Joining the crew shouting into the void asking for this small QOL improvement

**Comment by ybentz** [comment]
Another workaround to make it a one-click action. I wrote a tiny bookmarklet that simply adds the `w=1` query param to the existing url and reloads the page. Much faster than using the Github UI although still requires doing it for every PR so obviously not ideal.

Save this as the url of a normal bookmark and name it whatever you like. When you're on a GH PR simply click the bookmark to apply:
```
javascript:(function()%7Bconst url %3D new URL(window.location.href)%3Burl.searchParams.set("w"%2C "1")%3Bwindow.location.href %3D url.toString()%7D)()
```

**Reply by madisonscott** [reply]
!!!! Thank you for sharing! I love how simple this hack is and how obviously trustworthy it is 👏

**Comment by fmichaut-diff** [comment]
Having to turn this on every single PR is ridiculous.
We have appearance settings for the tab-space, default emoji skin color, but not for an actually useful setting like "Hide PR Whitespace by default" ?

Or put it in accessibility idk, my point is we already have user settings that auto-set default values for when we are writing comments on PRs, so why can't we have this one which already has a button for it. Its not like we are asking for a new feature, just for a setting to enable it by default !

**Comment by pixelscript** [comment]
I personally think it should be a project level setting, but persisting across PRs or making it a single click would be a good alternative.

**Comment by EskeRahn** [comment]
Though I use this hiding whitespace-differences all the time, I DO understand why they hesitate with making it _too_ easy to enable... 

We **all** want this to be a reliable tool, but we _must_ remember that _not all_ whitespaces are ignorable! So it _could_ give a nasty case where a major system have some breaking changes that are not seen due to this, reviewing a PR, and later for trying to debug the changes. And they risk getting a lot of bad publicity from such a case.

They want us to do it as an informed choice, knowing the risks, so we not just blindly use it!

A much better (and a LOT more complex solution) would be a **logical** compare, that could even catch simple renames of identifiers. But that would require the file to be interpreted/compiled in a suitable matter.

Currently we especially struggle in languages with block encapsulation chars like { and } that a single brace on a line with the right indentation is seen as a good match for synchronisation, where it seldom really is, and this is typically **the real culprit** in a compare looking garbled!

So perhaps a safer - yet text-only - option would be if it detected (and displayed) a block (including subblocks) with changed indentation as one difference only.

Or (as suggested elsewhere) when comparing texts, any line only containing one or more of  
 ( ) [ ] { } ;  
as the only thing on a line, should have this _logically appended_ to the previous line for doing the sync/compare. This would get rid of most of the false-sync hits between two version of some code.

**Reply by luchillo17** [reply]
I use it all the time but that's given I work with mostly JS, so whitespace is non significant in my workflow, but there are languages where whitespace is, specifically indentation based languages like Python, in that case I'm inclined to support a project based config for each user as mentioned by @pixelscript, so I can set it to true in any non indentation based project I frequently work with, and consciously keep it at false if I work in those special projects where whitespace is significant.

**Reply by EskeRahn** [reply]
@luchillo17, but it is not 100% correct that whitespaces are non-significant to JS. If they appear within a string they _could_ matter, especially if it not just used for displaying something. I know this is **rarely** the case, but the important thing here is to be aware that it _can_ have a significance, and that is why things are a bit more complex. But sure for the vast majority of the cases it will not matter.

My point is that it would be much better if the diff algorithm could display them as _likely_ insignificant, than to completely ignore them. And that is where better sync comes in.

I have (many years ago) wasted a lot of time due to EOL whitespaces in a string being stripped, that should not have been -  a later step in an several hours long process expected fixed length lines, without checking... And debugging I compared before/after ignoring whitespaces (back in DOS with FC /W) so the stripping-error was hidden, and I wasted a LOT of time reverting visible changes and rerunning, only to have it crash again.

**Reply by luchillo17** [reply]
Hmm, I thought spaces within strings were a given to be untouched, we're talking about whitespace changes within code, afaik I have never seen a `ignore whitespace changes` that would also ignore them within strings or multiline strings (backtick strings `` ` `` in JS), neither in GH or in GitKraken where I can set the diff tool to always ignore whitespaces.

**Reply by EskeRahn** [reply]
...To do that it has to do some rough compilation handling inline strings, here an arbitrary example, where some spaces matters and others do not. I do not know if they handle this correctly already.
```
   string xxx = $"somthing { foo } something { bar } something { ( useFirst ? "the first" : "a following" ) } something ";
```

**Reply by NicholasMTElliott** [reply]
Sure; but we _already_ have the feature to enable ignoring whitespace.  Whatever that behavior is, that's sufficient for the vast majority of us.  I've run into maybe one situation in the last 20 years where there was a bug because of whitespace being hidden in a code review.  However, I have to manually turn on this feature to ignore whitespace changes on several PRs _every day_.  If I'm in a situation where the whitespace finally matters... I will turn it off, do that review, and call it a day.

**Reply by luchillo17** [reply]
Indeed, the issue is not if it works properly, the issue is that it exists and we want to enable it either globally or by project.

**Reply by EskeRahn** [reply]
Do not get me wrong, it would certainly be better with even a global switch than what we got now. I'm just saying that I would hesitate too if I had to decide, as this is just a workaround (even with caveats), and not a _solution_ to the issue. And I would suggest that the workaround was bound to filetypes, as this would rule out the worst caveats.

**Comment by mark-langer** [comment]
@willsmythe since you were involved in the [original thread](https://github.com/orgs/community/discussions/5486): Is there anything you guys could do to raise the priority on this?

**Comment by logic01** [comment]
Wow, 7 years later and we all still dream.

**Comment by cduflo** [comment]
Copying over from the original thread asking for this feature, I wrote a small chrome extension that handles this as it doesn't seem like GH is going to provide this feature - https://chrome.google.com/webstore/detail/git-well-soon/ehpeaofieafibmhiagianfjjblpnmbdo

https://github.com/orgs/community/discussions/5486#discussioncomment-4656459

**Reply by cduflo** [reply]
It had been awhile since I wrote this, so I went back to rewrite it to avoid needing special permissions :)

**Reply by fmichaut-diff** [reply]
Thanks for the extension, works great. However, could you do make it so that it scrapes links on pages and update any relevant github link with the `w=1` param ? (or maybe intercept onClick events on links, and inspect their href tag and handle the redirection yourself if the link matches github)

This would avoid having to load the page 2 times, cause currently your extension looks at the url so once the browser has already navigated there, and then you change it. Granted it is still quite fast, but you can see the stuttering of the page when the url change happens.

**Reply by cduflo** [reply]
Thanks for taking the time to send some feedback, @fmichaut-diff ! Unfortunately what you are proposing - while a great idea - requires manifest host_permissions - which gives users a warning like: “Read and change all your data on the websites you visit”. I just recently refactored to remove that permission to put users more at ease, but it takes away the extension's ability to implement your proposal. I'll add it as an issue on the project and maybe there's a loophole I'm overlooking https://github.com/cduflo/gitwellsoon/issues

**Reply by acidoxee** [reply]
I'd also like to have reduced flicker if possible. With Refined GitHub also installed, PR pages sometimes reload two or three times when navigated to, with quite a lot of layout shift. And since GitHub isn't really loading fast these days, it's really not a good experience 😕

**Reply by cduflo** [reply]
Fair enough - seems like there's more uptake for this mode than I thought. I've had some more ideas overnight, I'll keep this issue updated as I have time in the coming week to address this. Thanks for the input 😃 
https://github.com/cduflo/gitwellsoon/issues/3

**Reply by cduflo** [reply]
v1.1.0.0 of the extension is out now, hopefully you notice the improvement. It was achievable without enhancing the permissions requirements, which was nice! Please let me know if you encounter any adverse behavior post-update: https://github.com/cduflo/gitwellsoon/issues

Please give the extension some stars if you think it deserves them https://chromewebstore.google.com/detail/git-well-soon/ehpeaofieafibmhiagianfjjblpnmbdo 🌟

**Reply by acidoxee** [reply]
I can confirm the double reload is gone, there's just the one that Refined GitHub introduces now (which I might opt out of despite the benefits since it's really bothering me). No issues with it until now. Thanks a bunch!

**Comment by ken-ailevate** [comment]
Another vote to persist 'Hide whitespace' the same way as 'Diff view'.

**Comment by taylorham** [comment]
Is anyone from Github out there? Anybody listening to this at all? I've been assuming this was on the radar for implementation for more than half a decade given how crazy easy it would be to persist just like the "Split vs. Unified" view option.

How are we still begging for this? Why is no one providing any kind of response?

**Comment by ozturkkl** [comment]
+1 here, it's incredibly frustrating for something like this to go back to the default setting and having to change it everytime there's a new PR to reveiw.

Why is this still not fixed after YEARS of ask...

**Comment by roryap** [comment]
I just spent a while looking at a changed C# file which was showing hundreds upon hundreds of changes--it was chaos.  Then I hid whitespace and realized the only change was to convert from block-scoped namespace to file-scoped, which can be done with a quick action to any C# file without even knowing that it could cause reviewers potentially hours of befuddlement.

Please fix this.

**Comment by pmarshalla** [comment]
+1 here , this simple feature still hasnt been added ? its very frustrating having to do this on EVERY PR!!!! Please change.

**Comment by luchillo17** [comment]
I'm using the Refined Github extension and it helps that it puts the `Hide whitespaces` button directly on top of the diff, saves me at least 1 click and the popover animation time...
![image](https://github.com/user-attachments/assets/0d645b86-8ca8-406c-b849-6f204c418fd0)

**Comment by ardicoetzee** [comment]
There is a slight improvement. But I don't know why they didn't just fix it properly when they made this change.

Now, if you check "Hide whitespace", then at least it remains that way **for that project**.

**Comment by ggalmazor** [comment]
Since August 2021 (when the [original discussion](https://github.com/orgs/community/discussions/5486) for this was created), I've been involved in more than 3.3k PRs on GitHub, and I've always required enabling `Hide whitespace` either manually or through a browser extension. Being able to set this as a default for my projects/organizations/user would save me a lot of time in the aggregate.

The available browser extensions I've tried for this purpose don't eliminate the pain of this interaction, often requiring page reloads. 

This is a feature in many GUIs:

**Visual Studio Code**
<img width="1267" height="270" alt="image" src="https://github.com/user-attachments/assets/f5735c0a-dcaa-4d29-b704-9ccb89c5bfec" />

**JetBrains IDEs**
<img width="681" height="189" alt="Screenshot 2025-07-26 at 09 57 41" src="https://github.com/user-attachments/assets/13617a45-f354-4da4-a9dc-50d2e61709ad" />


Please provide an option to set a default for the `Hide whitespace` diff option for a scope wider than a single PR, per project, organization, or user. An option to set the default per user would make the most sense to me personally, but any additional scopes for this default would be a considerable improvement at this point.

**Comment by andrea-cassioli-maersk** [comment]
Come on github, please fix this!

**Comment by ktarbet** [comment]
I'm now trained to first click 'hide white space'. 

However, when a larger file is involved in a PR  I click the link: "Large diffs are not rendered by default." And white space is shown, regardless of current white space setting.

**Comment by PwrSrg** [comment]
Please fix this. 🙏🏻

**Comment by cody-owens** [comment]
This is a very important quality of life capability that other competitors have. Please add this basic feature.

**Comment by learn2reid** [comment]
Please add this!

**Comment by Aynekko** [comment]
Why is this still not implemented yet! It's a simple option and everybody wants it, PLEASE! I want to view commits with whitespace hidden by default. Please add a check in user profile for this, or something. I beg you!

**Comment by reneleonhardt** [comment]
Interestingly enough, the GitHub diff doesn't show Git whitespace errors anymore (file contained no trailing line break before the commit)... maybe it's hiding whitespace a bit too much now 😄

**Comment by s-robertson** [comment]
I was so tired of having to turn off whitespace changes on every PR that I wrote a Chrome extension that just does it for every diff page on GitHub. _Please_ let us just choose the default value for this for ALL PRs so I can stop this madness.

https://github.com/s-robertson/github-ws

**Reply by abedolinger** [reply]
worth mentioning this works for firefox too!

**Comment by petevb** [comment]
I agree it'd be great to set this by default. Meantime, you can add this as a bookmarklet: 
`javascript:(function(){var u=new URL(location.href);if(u.searchParams.get('w')==='1'){location.href=u.toString();return;}u.searchParams.set('w','1');location.href=u.toString();})();`

**Comment by Entroper** [comment]
IMO, the pull request page is the single most important page on all of GitHub.  It's where the act of collaboration actually happens in the repository.  The diff view is the most important tab on that page.  And this is a really irritating thorn in the side of everyone looking at a diff.

**Comment by PfunnyGuy** [comment]
PLEASE add this.  Right now the existing setting doesn't work if you have a large diff.  I click "Load Diff" link because "Large diffs are not rendered by default." and it does NOT ignore whitespace.  Only what is actually visible when I click the option gets whitespace ignored.  This implementation is so incredibly lazy!  It makes reviewing some files take an order of magnitude longer.  You're a tool maker, please improve your tool!

**Comment by danny-pflughoeft** [comment]
Absolutely wild this request has been open so long. Do Github devs not use their own product?

**Comment by 3htomit** [comment]
@willsmythe, [as you answered the previous thread](https://github.com/orgs/community/discussions/5486#discussioncomment-1541120), can you please give us an update on this long running and highly expected request ?

**Comment by dtitov** [comment]
Bump. It's mind-blowing how such a simple feature, requested by so many people, is ignored for such a long time...

**Comment by youssef-lr** [comment]
Workaround in the meantime https://chromewebstore.google.com/detail/github-whitespace/fnpkdafamnbjoldglihkjjdicofghccm

**Reply by mateusfccp** [reply]
Imagine using Google Chrome.

**Reply by youssef-lr** [reply]
You can install it on any chromium fork. I use Brave.

**Comment by mateusfccp** [comment]
It's just stupid that this was not yet implemented. The original post is from 2021, and this is from 2023 and we have no answer. Can't expect much from Microslop.

**Reply by EskeRahn** [reply]
What is stupid here is not that it is not implemented, but that they do not bother to try to explain WHY they do not implement it.

It CERTAINLY would most often be VERY convenient, but as I tried to explain above, I too would hesitate implementing it blindly, as it MIGHT make people overlook breaking changes that an ignore white-spaces might have missed as crucial. It requires understanding of the text to know if white-spaces matter or not. Maybe an AI variant can give us that distinction between important and unimportant white-spaces.

ADD:
When we _actively_ have to choose to hide the white-spaces, that nuisance is a clear choice **we** make, so that they can not in any way be considered liable for something that breaks due to a change that was not shown in the diff-view. So my guess would be it is a lot about _cover-our-a..es_ that is in play here.

**Reply by luchillo17** [reply]
For the most part white-spaces only matter in 2 situations, inside strings (template strings, plain strings), and in indentation based languages like Python, other than that I don't see whitespaces as significant in a diff, especially if you have auto format tools like prettier taking care of the white spaces.

**Comment by joshuaandrewhoffman** [comment]
Absolutely floored that this feature is STILL not implemented in the year-of-our-lord-2026 despite the overwhelming demand by the community at large for over five years.

Just let us write the damn feature for you at this point.... It's a persisted settings toggle that changes a query string for goodness sake

**Comment by vamnguyen** [comment]
Thanks Github :)
